sdk/apimanagement/Azure.ResourceManager.ApiManagement/src/Customized/ApiResource.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.ApiManagement.Models;
+
+namespace Azure.ResourceManager.ApiManagement
+{
+    public partial class ApiResource
+    {
+        private static readonly ResourceType ResourceTypeWithGateway = "Microsoft.ApiManagement/service/gateways/apis";
+
+        private static readonly ResourceType ResourceTypeWithProduct = "Microsoft.ApiManagement/service/products/apis";
```

Comment:
I am kind of confused about this resource.
The static property of this resource will always return `Microsoft.ApiManagement/service/apis`, does this mean that when I call this API
`/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/apis/{apiId}`, sometimes it returns an ID of `Microsoft.ApiManagement/service/apis`, sometimes it does not and returns `Microsoft.ApiManagement/gateways/apis`?

Reply:
And our `CreateResourceIdentifier` method in this resource will also always return an ID of type `Microsoft.ApiManagement/service/apis`

Reply:
If you call `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/apis/{apiId}`, it returns an ID of `Microsoft.ApiManagement/service/apis`. But we can also add the created api to a gateway or a product, in this case, the response still represents an `ApiResource`, but the id will be [/subscriptions/bab08e11-7b12-4354-9fd1-4b5d64d40b68/resourceGroups/Api-Default-CentralUS/providers/Microsoft.ApiManagement/service/sdktestservice/gateways/gatewayid8779/apis/echo-api](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/apimanagement/Microsoft.Azure.Management.ApiManagement/tests/SessionRecords/GatewayTests/CreateListUpdateDelete.json#L667). That's the reason that I extended the scope of this validation.

Reply:
We don't have this kind of resource, and I didn't test the scenario that you mentioned but I think it might work. Actually I think the better choice here is to return the data instead of introducing this kind of customization.